internal/kafka/internal/services/kafka.go

```diff
@@ -1150,6 +1214,73 @@ func buildManagedKafkaCR(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 	return managedKafkaCR, nil
 }
 
+// buildReservedManagedKafkaCR buklds a Reserved Managed Kafka CR.
+// The ID, K8s object ID, K8s namespace and PlacementID are all set to
+// the provided kafkaID.
+func (k *kafkaService) buildReservedManagedKafkaCR(kafkaID string, instanceType string, desiredStrimziVersion api.StrimziVersion, kafkaConfig *config.KafkaConfig) (*managedkafka.ManagedKafka, *errors.ServiceError) {
+	// Reserved instances always make use of the streaming base unit which is x1.
+	// For now we hardcode it but there might be a better alternative to it.
```

Comment:
We won't be doing the above, at least in this first version.
Partly because, we need to have the support for higher sizes and for it to be in stable maturity state.